### PR TITLE
Remove legacy token endpoints

### PR DESCRIPTION
**Pull request by otherchen**
1. Removed the legacy token implementations
2. Removed the tests associated with the legacy tokens
3. Updated tests to no longer check for the `removed` flag in the `/item/remove` response
4. Updated /payment/get test to no longer check for payment_token or payment_token_expiration_time

### CODE REVIEW

File: tests/integration/test_payment_initiation.py [hunk 1]
```diff
@@ -20,22 +22,29 @@ def payments_after_recipient_creation(client, recipient_id):
...
     assert response['status'] is not None
 
-    # create payment token
-    response = client.PaymentInitiation.create_payment_token(
-        payment_id,
-    )
-    assert response['payment_token'] is not None
-    assert response['payment_token_expiration_time'] is not None
+    # create link token
+    response = client.LinkToken.create({
```
Comment (skylarmb) [inline tests/integration/test_payment_initiation.py:26]: is there another test for `client.LinkToken.create` somewhere else already?

**Review by skylarmb — APPROVED** [review]

File: tests/integration/test_payment_initiation.py (hunk 1 above)
Comment (otherchen) [reply tests/integration/test_payment_initiation.py:26]: Yep! There should already be a test for `client.LinkToken.create`

File: tests/integration/util.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def create_client():
         client_id=os.environ['CLIENT_ID'],
         secret=os.environ['SECRET'],
         environment='sandbox',
-        api_version="2019-05-29",
+        api_version="2020-09-14",
```
Comment (skylarmb) [inline tests/integration/util.py:14]: is it not pinned anywhere else? do we allow clients to pass api version?

**Review by skylarmb — APPROVED** [review]

File: tests/integration/util.py (hunk 2 above)
Comment (otherchen) [reply tests/integration/util.py:14]: Yeah, we allow clients to pass api versions in currently for this clib :/

**Review by cgfarmer4 — APPROVED** [review]

**Pull request merged**